src/consensus/aft/raft.h

```diff
@@ -2214,7 +2320,19 @@ namespace aft
 
       if (is_elected)
       {
-        become_leader();
+        switch (state->leadership_state)
+        {
+          case ccf::kv::LeadershipState::PreVoteCandidate:
+            become_candidate();
+            break;
+          case ccf::kv::LeadershipState::Candidate:
+            become_leader();
+            break;
+          default:
+            throw std::logic_error(
+              "add_vote_for_me() called while not a pre-vote candidate or "
```

Comment:
I do wonder if we should be strict about using enum values consistently in error messages. It is verbose, but also pleasantly greppable.